### DISCUSSION TITLE
Have enemies "shift" up walls

### DISCUSSION THREAD

**Discussion by JimmyThrush** [Techokami/SonicWorldsNext#132] [Q&A] [answered]
I haven't looked at the assembly code for myself, but it seems like enemies in the Genesis Sonic games had no "collision hitboxes" to determine their movement, but only sensors to determine what they did in response to coming in contact with terrain. Take, for example, this footage of the Burrobots from Sonic 1's Labyrinth Zone. They'll stop at sheer walls, but for these 32px blocks, they shift upwards, leading me to believe their wall sensors are placed higher than 32px specifically for this behavior.
[test.webm](https://github.com/Techokami/SonicWorldsNext/assets/123826402/54065790-dc52-4cf2-b8c6-f73122f78e59)
This is also how other more mobile enemies like Rollers deal with shifts in terrain like this.

How could I go about implementing this kind of behavior? PhysicsObject.gd's push_vertical() appears to handle this sort of interaction for the player, but I can't find the conditions for it being triggered. I'd like to add support for this behavior in my project's EnemyBase.gd to have Genesis-accurate enemy behavior.

**Comment by Renhoex** [comment, marked as answer]
You're probably going to want to do something simpler, take a look at the flicky and scattered ring objects for reference.
I'd have a raycast that points down from the top of the enemies body, and have it shift the enemy to the collision point plus the offset from the center if it detects a collision

**Reply by JimmyThrush** [reply]
Thanks a ton; it works perfectly.
